lib/rules/logical-assignment-operators.js

```diff
@@ -0,0 +1,473 @@
+        },
+
+        schema: {
+            type: "array",
+            oneOf: [{
+                items: [
+                    { const: "always" },
+                    {
+                        type: "object",
+                        properties: {
+                            enforceForIfStatements: {
+                                type: "boolean"
+                            }
+                        },
+                        additionalProperties: false
+                    }
+                ],
+                minItems: 0, // 0 for allowing passing no options
+                maxItems: 2
+            }, {
+                items: [{ const: "never" }],
+                minItems: 1,
+                maxItems: 1
+            }]
+        },
+        fixable: "code",
+        // eslint-disable-next-line eslint-plugin/require-meta-has-suggestions -- Does not detect conditional suggestions
+        hasSuggestions: true,
+        messages: {
+            assignment: "Assignment (=) can be replaced with operator assignment ({{operator}}).",
+            useLogicalOperator: "Convert this assignment to use the operator {{ operator }}.",
+            logical: "Logical expression can be replaced with an assignment ({{ operator }}).",
+            convertLogical: "Replace this logical expression with an assignment with the operator {{ operator }}.",
+            if: "'if' statement can be replaced with a logical operator assignment with operator {{ operator }}.",
+            convertIf: "Replace this 'if' statement with a logical assignment with operator {{ operator }}.",
+            unexpected: "Unexpected logical operator assignment ({{operator}}) shorthand.",
+            separate: "Separate the logical assignment into an assignment with a logical operator."
+        }
+    },
+
+    create(context) {
+        const mode = context.options[0] === "never" ? "never" : "always";
+        const checkIf = mode === "always" && context.options.length > 1 && context.options[1].enforceForIfStatements;
+        const sourceCode = context.getSourceCode();
+        const isStrict = context.getScope().isStrict;
+
+        /**
+         * Returns false if the access could be a getter
+         * @param {ASTNode} node Assignment expression
+         * @returns {boolean} True iff the fix is safe
+         */
+        function cannotBeGetter(node) {
+            return node.type === "Identifier" &&
+                   (isStrict || !isInsideWithBlock(node));
+        }
+
+        /**
+         * Check whether only a single property is accessed
+         * @param {ASTNode} node reference
+         * @returns {boolean} True iff a single property is accessed
+         */
+        function accessesSingleProperty(node) {
+            if (!isStrict && isInsideWithBlock(node)) {
+                return node.type === "Identifier";
+            }
+
+            return node.type === "MemberExpression" &&
+                   baseTypes.has(node.object.type);
```

Comment:
```suggestion
            return node.type === "MemberExpression" &&
                    baseTypes.has(node.object.type) &&
                    node.property.type !== "MemberExpression" &&
                    node.property.type !== "ChainExpression";
```

Example: 

```js
/* eslint logical-assignment-operators: ["error", "always", { enforceForIfStatements: true }] */

if (a[b.c]) a[b.c] = 5; // a[b.c] &&= 5;
```

In the original code, `b.c` can be evaluated twice, while in the fixed code it's always only once, so this should be a suggestion (not autofix).